lib/util/http/index.ts

```diff
@@ -203,9 +215,14 @@ export class Http<Opts extends HttpOptions = HttpOptions> {
     }
 
     try {
-      const res = await resPromise;
+      const res = memCacheKey
+        ? cloneResponse(await resPromise)
+        : await resPromise;
       res.authorization = !!options?.headers?.authorization;
-      return cloneResponse(res);
+      if (etagCache && res.statusCode === 304) {
+        res.body = clone(etagCache.data);
```

Comment:
While response can be cloned conditionally, it's better to always clone data obtained from parameter just in case. One day it can be from package case, another one it will come from in-memory map.